template/9-design.md

```diff
@@ -1,56 +1,193 @@
 # Design and Implementation
 
 ## Frontend
+### Implementation Framework
+The app will be developed in Kotlin, optimized for Android, utilizing the Model-View-ViewModel (MVVM)architecture. This approach ensures a clear separation of concerns, making the management of the app’s UI and business logic more efficient and maintainable.
+### Key Libraries and Languages
+- **Kotlin**: Primary language for our Android development
+- **Hilt**: Dependency injection library for managing dependencies and improving code modularity.
+- **Material3**: Used for all graphic components
+- **Jetpack Compose**: Used for dynamic UI rendering.
+- **Firebase**: Backend-as-a-Service platform providing authentication, database, and cloud storage.
+- **Android Camera Image Analysis**: Utilized for integrating camera functionalities required for photo capture during friend requests.
+- **Dagger Assisted Factory**: Design pattern to create objects, particularly useful in the ViewModel creation process, ensuring that instances are correctly initialized with necessary dependencies.
+- **ZXing**: Library to encode information to QR codes and decode QR codes to information
+### Dynamic UI Rendering Strategy
+The app will leverage Jetpack Compose for dynamic UI rendering. Compose enables the creation of UI components declaratively, allowing for real-time updates and modifications based on data changes.
+- **Composable Functions**: These will be used to define each screen and UI component, enabling reusable and modular UI elements.
+- **State Management**: ViewModels will manage the state, ensuring the UI is always in sync with the underlying data model.
+### Essential Screens
+#### 1 Home Screen
+- **Features**: Displays all events as well as events the user has registered to in either list or map view
+- **Components**: Event list, Upcoming events, search-filter, list/map view button
```

Comment:
perhaps mention nav bar?

Reply:
Added it as an extra Essential Screen and explained that it was at the bottom of all other screens